src/psyclone/tests/psyad/transformations/test_preprocess.py

```diff
@@ -129,3 +129,31 @@ def test_preprocess_matmul(tmpdir, fortran_reader, fortran_writer):
     result = fortran_writer(psyir)
     assert result == expected
     assert Compile(tmpdir).string_compiles(result)
+
+
+def test_preprocess_arrayrange2loop(tmpdir, fortran_reader, fortran_writer):
+    '''Test that the preprocess script replaces assignments that contain
+    arrays that use range notation with equivalent code that uses
+    explicit loops.
+
+    '''
+    code = (
+        "program test\n"
+        "real :: a(10), b(10), c(10)\n"
+        "a(:) = b(:) * c(:)\n"
+        "end program test\n")
+    expected = (
+        "program test\n"
+        "  real, dimension(10) :: a\n"
+        "  real, dimension(10) :: b\n"
+        "  real, dimension(10) :: c\n"
+        "  integer :: idx\n\n"
+        "  do idx = LBOUND(a, 1), UBOUND(a, 1), 1\n"
```

Comment:
I thought we had code that was clever and would work out what the lower and upper bounds were if possible (as it is in this case). Am I imagining things @sergisiso ?

Reply:
No we don't do that yet. We use bounds information if it is part of the notation (e.g: a(2:n)) and we use it if we have config information in the nemo api implementation but we do not look back at the declaration.